rotkehlchen/chain/aggregator.py

```diff
@@ -1660,35 +1691,40 @@ def add_accounts_to_all_evm(
-        eth_contract_addresses: list[ChecksumEvmAddress] = []
+        evm_contract_addresses: list[tuple[SUPPORTED_EVM_EVMLIKE_CHAINS_TYPE, ChecksumEvmAddress]] = []  # noqa: E501
 
         for account in accounts:
             existed_accounts += [(chain, account) for chain in SUPPORTED_EVM_EVMLIKE_CHAINS if account in self.accounts.get(chain)]  # noqa: E501
             # Distinguish between contracts and EOAs
-            if self.is_contract(account, SupportedBlockchain.ETHEREUM):
-                added_chains, _ = self.track_evm_address(account, [SupportedBlockchain.ETHEREUM])
-                if len(added_chains) == 1:  # Is always either 1 or 0 since is only for ethereum
-                    added_accounts.append((SupportedBlockchain.ETHEREUM, account))
-                    eth_contract_addresses.append(account)
-            else:
-                chains_to_check = [x for x in SUPPORTED_EVM_EVMLIKE_CHAINS if account not in self.accounts.get(x)]  # noqa: E501
-                new_accounts, new_failed_accounts, had_activity = self.check_chains_and_add_accounts(  # noqa: E501
-                    account=account,
-                    chains=chains_to_check,
-                )
+            chains_to_check = [x for x in SUPPORTED_EVM_EVMLIKE_CHAINS if account not in self.accounts.get(x)]  # noqa: E501
+            chains_with_valid_addresses = []
+            for chain in chains_to_check:
+                if chain in (SupportedBlockchain.AVALANCHE, SupportedBlockchain.ZKSYNC_LITE):
+                    chains_with_valid_addresses.append(chain)
+                    continue
+
+                if self.is_safe_proxy_or_eoa(address=account, chain=chain):  # type: ignore  # the chain is a supportedblockchain here
+                    chains_with_valid_addresses.append(chain)
+                else:
+                    evm_contract_addresses.append((chain, account))
+
+            new_accounts, new_failed_accounts, had_activity = self.check_chains_and_add_accounts(
+                account=account,
+                chains=chains_with_valid_addresses,
+            )
 
-                if had_activity is True:
-                    added_accounts += new_accounts
-                    failed_accounts += new_failed_accounts
-                elif had_activity is False and len(chains_to_check) != 0:
-                    no_activity_accounts += [(chain, account) for chain in chains_to_check]
+            if had_activity is True:
+                added_accounts += new_accounts
+                failed_accounts += new_failed_accounts
+            elif had_activity is False and len(chains_with_valid_addresses) != 0:
+                no_activity_accounts += [(chain, account) for chain in chains_with_valid_addresses]
 
         return (
             added_accounts,
             existed_accounts,
             failed_accounts,
             no_activity_accounts,
-            eth_contract_addresses,
+            evm_contract_addresses,
```

Comment:
you need to modify the docstring in `detect_evm_accounts` right below my comment